packages/arb-ts/integration_test/testHelpers.ts

```diff
@@ -60,3 +83,77 @@ export const warn = (text: string) => {
   console.log(chalk.red(`WARNING: ${text}`))
   console.log()
 }
+
+export const instantiateRandomBridge = () => {
+  const testPk = utils.formatBytes32String(Math.random().toString())
+  prettyLog(
+    `Generated wallet, pk: ${testPk} address: ${new Wallet(testPk).address} `
+  )
+  return instantiateBridge(testPk)
+}
+
+const _preFundedWallet = new Wallet(process.env.DEVNET_PRIVKEY as string)
+const _preFundedL2Wallet = new Wallet(process.env.DEVNET_PRIVKEY as string)
+
+console.warn('using prefunded wallet ', _preFundedWallet.address)
+
+export const fundL1 = async (bridge: Bridge) => {
+  const testWalletAddress = await bridge.l1Bridge.getWalletAddress()
+  const preFundedWallet = _preFundedWallet.connect(bridge.l1Provider)
+  const res = await preFundedWallet.sendTransaction({
+    to: testWalletAddress,
+    value: preFundAmount,
+  })
+  const rec = await res.wait()
+  prettyLog('Funded L1 account')
+}
+export const fundL2 = async (bridge: Bridge) => {
+  const testWalletAddress = await bridge.l2Bridge.getWalletAddress()
+  const preFundedL2Wallet = _preFundedL2Wallet.connect(bridge.l2Provider)
+  const res = await preFundedL2Wallet.sendTransaction({
+    to: testWalletAddress,
+    value: preFundAmount,
+  })
+  const rec = await res.wait()
+  prettyLog('Funded L2 account')
+}
+
+export const tokenFundAmount = BigNumber.from(2)
+export const fundL2Token = async (bridge: Bridge, tokenAddress: string) => {
+  try {
+    const testWalletAddress = await bridge.l2Bridge.getWalletAddress()
+    const preFundedL2Wallet = _preFundedL2Wallet.connect(bridge.l2Provider)
+    const l2Address = await bridge.getERC20L2Address(tokenAddress)
+    const testToken = TestERC20__factory.connect(l2Address, preFundedL2Wallet)
+
+    const res = await testToken.transfer(testWalletAddress, tokenFundAmount)
+    const rec = await res.wait()
+
+    const result = rec.status === 1
+    result && prettyLog('Funded L2 account w/ tokens')
+
+    return result
+  } catch (err) {
+    console.warn('err', err)
+
+    return false
+  }
+}
+
+export const wait = (ms = 0) => {
+  return new Promise(res => setTimeout(res, ms))
+}
+
+export const skipIfMainnet = (() => {
+  let chainId = ''
+  return async (testContext: Mocha.Context) => {
+    if (!chainId) {
+      const { l1Network } = await instantiateRandomBridge()
+      chainId = l1Network.chainID
+    }
+    if (chainId === '1') {
+      console.log("You're writing to the chain on mainnet lol stop")
+      testContext.skip()
```

Comment:
why would we skip a particular test/context instead of just halting altogether?

Reply:
i want to leave room for some tests to be acceptable to run still (i.e., sanity.test) so we skip on a case by case basis